From: Florent Kermarrec <florent@enjoy-digital.fr>
Subject: [PATCH] generic: Respect firmware address overrides from the environment

Buildroot passes BR2_TARGET_OPENSBI_ADDITIONAL_VARIABLES in the make
process environment. Preserve these explicit settings instead of
replacing the LiteX addresses with the generic platform defaults.

Signed-off-by: Florent Kermarrec <florent@enjoy-digital.fr>

--- a/platform/generic/objects.mk
+++ b/platform/generic/objects.mk
@@ -22,17 +22,17 @@
 platform-objs-y += platform_override_modules.o
 
 # Blobs to build
-FW_TEXT_START=0x80000000
+FW_TEXT_START?=0x80000000
 FW_DYNAMIC=y
 FW_JUMP=y
 ifeq ($(PLATFORM_RISCV_XLEN), 32)
   # This needs to be 4MB aligned for 32-bit system
-  FW_JUMP_ADDR=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x400000)))
+  FW_JUMP_ADDR?=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x400000)))
 else
   # This needs to be 2MB aligned for 64-bit system
-  FW_JUMP_ADDR=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x200000)))
+  FW_JUMP_ADDR?=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x200000)))
 endif
-FW_JUMP_FDT_ADDR=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x2200000)))
+FW_JUMP_FDT_ADDR?=$(shell printf "0x%X" $$(($(FW_TEXT_START) + 0x2200000)))
 FW_PAYLOAD=y
 ifeq ($(PLATFORM_RISCV_XLEN), 32)
   # This needs to be 4MB aligned for 32-bit system
